datadog_checks_base/datadog_checks/checks/base.py

```diff
@@ -330,3 +339,34 @@ def _get_requests_proxy(self):
             proxies['no'] = proxies.pop('no_proxy')
 
         return proxies if proxies else no_proxy_settings
+
+    @staticmethod
+    def _get_statistic_name_from_method(method_name):
+        return method_name[4:] if method_name.startswith('get_') else method_name
+
+    @staticmethod
+    def _collect_internal_stats(methods=None):
```

Comment:
Let's please put these methods in `AgentMetrics` class instead.

Reply:
They used to be in the AgentCheck class, so I moved them there

Reply:
Incidentally this means the agent metrics check has been broken for a while :-/

Reply:
https://github.com/DataDog/dd-agent/blob/master/checks/__init__.py#L741-L770